app/controllers/application_controller.rb

```diff
@@ -9,4 +11,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  # Class accessor for an instance of RedirectMap
+  def self.redirect_map
+    @redirect_map ||= RedirectMap.instance
+  end
+
+  # Instance accessor for the redirect map. This is used in the :before_action
+  # hook.
+  def redirect_map
+    self.class.redirect_map
+  end
+
+  if Rails.env != 'test'
+    redirect_map.load Rails.root.join('config', 'redirect_map.yml')
+  end
```

Comment:
I don't have any benchmarks. I just made a sample redirect map file containing 46,000 entries mapping a sample URL "catalog/wpna-2ab613-interview-with-kenneth-adelman-1987" to itself. I think this would be a good rough estimate at the size of the file we'd be dealing with. The size is only 5.1MB, which doesn't take that long to load. Let's leave it, and if it's a problem, address it later.